benchmark/buffers/buffer-compare-offset.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+var common = require('../common.js');
+
+var bench = common.createBenchmark(main, {
+  method: ['offset', 'slice'],
+  size: [16, 512, 1024, 4096, 16386],
+  millions: [1]
+});
+
+function compareUsingSlice(b0, b1, size, iter) {
+  const len = size >> 2;
+  for (var i = 0; i < iter; i++)
```

Comment:
In here we should probably force optimization of `Buffer.compare()`, `b0.slice()`, and `b1.slice()`. See the `path` benchmarks on how to do this.